### PR TITLE
flush stdin before asking for input

### PR DESCRIPTION
**Pull request by dirk-thomas**
Same as ros-infrastructure/bloom#249.

@wjwwood Please review.

### CODE REVIEW

**Comment by wjwwood** [comment]
So I noticed in bloom that sometimes the messages which are printed to stdout before one of these prompts never gets displayed. I'm not sure I can reproduce it, but it has definitely happened.

**Comment by dirk-thomas** [comment]
But do you think it is related to the stdin flushing?

**Comment by wjwwood** [comment]
I do actually, this is an example of what I am talking about:

```
==> Generating pull request to distro file located at 'https://raw.github.com/ros/rosdistro/master/hydro/distribution.yaml'
Would you like to add source information for this repository? [Y/n]? n
Would you like to add a maintenance status for this repository? [Y/n]?
Please enter a maintenance status.
Valid maintenance statuses:
- developed: active development is in progress
Status:
```

There should be two other `-` prefixed options before the `Status:` line.

Maybe we should flush stdout before truncating the stdin? I can try that locally and see if that make a difference.

File: bin/catkin_prepare_release [hunk 1]
```diff
@@ -49,6 +50,15 @@ def prompt_continue(msg, default):
         print(fmt("@{yf}Response '@{boldon}%s@{boldoff}' was not recognized, please use one of the following options: %s" % (response, ', '.join([('@{boldon}%s@{boldoff}' % x) for x in ['y', 'Y', 'n', 'N']]))), file=sys.stderr)
 
 
+def _flush_stdin():
+    try:
+        from termios import tcflush, TCIOFLUSH
+        tcflush(sys.stdin, TCIOFLUSH)
```
Comment (dirk-thomas) [inline bin/catkin_prepare_release]: Actually shouldn't this line be:

```
tcflush(sys.stdin, TCIFLUSH)
```

**Comment by dirk-thomas** [comment]
@wjwwood Can you please give it a try with my previous comment on changing `TCIOFLUSH` to `TCIFLUSH`?

**Comment by wjwwood** [comment]
I'll give it a try now.

**Comment by wjwwood** [comment]
+1

**Pull request merged**